#! /usr/bin/env python

# Control parameters for turtle simulation
PACKAGE='io_turtle'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name         Type        Level   Description                Default Min  Max
gen.add("k_rep",     double_t,   0,      "Repulsive Gain",          1.0,    0,   10)
gen.add("k_att",     double_t,   0,      "Attractive Gain",         2,      0,   10)
gen.add("k_theta",   double_t,   0,      "Angle Gain - P",          5.0,    0,   10)
gen.add("f_att_max", double_t,   0,      "Max Attractive Force",    4,      0,   10)
gen.add("v_max",     double_t,   0,      "Max Velocity",            5,      0,   10)
gen.add("omega_max", double_t,   0,      "Max Angular velocity",    2,      0,   10)

exit(gen.generate(PACKAGE, "io_turtle", "Control"))
